From 6bcf95fd0f32cf5e8b1ddbe17b14d9ad049bded8 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 27 Oct 2019 14:10:08 +0100
Subject: [PATCH] meson: fix cpu detection

Use host_machine.cpu_family() instead of cpu(). Only the former
provides a stable/defined set of values.

Also don't error out on some arches for no good reason.

This fixes the build on 32bit Windows.
---
 meson.build | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/meson.build b/meson.build
index af133a0f1..540498921 100644
--- a/meson.build
+++ b/meson.build
@@ -73,23 +73,21 @@ if os_osx and cc.get_id() != 'clang'
 endif
 
 
-host_cpu = host_machine.cpu()
-if   host_cpu.startswith('i') and host_cpu.endswith('86')
+host_cpu_family = host_machine.cpu_family()
+if   host_cpu_family == 'x86'
   have_x86 = true
   config.set10('ARCH_X86',    true)
-elif host_cpu == 'x86_64'
+elif host_cpu_family == 'x86_64'
   have_x86 = true
   config.set10('ARCH_X86',    true)
   config.set10('ARCH_X86_64', true)
-elif host_cpu == 'ppc' or host_cpu == 'powerpc'
+elif host_cpu_family == 'ppc'
   have_ppc = true
   config.set10('ARCH_PPC',    true)
-elif host_cpu == 'ppc64' or host_cpu == 'powerpc64'
+elif host_cpu_family == 'ppc64'
   have_ppc = true
   config.set10('ARCH_PPC',    true)
   config.set10('ARCH_PPC64',  true)
-else
-  error('Unknown host architecture')
 endif
 
 ################################################################################
